File: Statoil/8000.Prav_ModelEnsembling.py
# -*- coding: utf-8 -*-
"""
Created on Thu Dec 14 21:37:14 2017

@author: SriPrav
"""

import pandas as pd

inDir = 'C:/Users/SriPrav/Documents/R/35Statoil'

fold1 = pd.read_csv(inDir + "/submissions/Prav.vgg03.fold1-test.csv")
fold2 = pd.read_csv(inDir + "/submissions/Prav.vgg03.fold2-test.csv")
fold3 = pd.read_csv(inDir + "/submissions/Prav.vgg03.fold3-test.csv")
fold4 = pd.read_csv(inDir + "/submissions/Prav.vgg03.fold4-test.csv")

test = pd.merge(fold1, fold2, on="id",how="left")
test = pd.merge(test, fold3, on="id",how="left")
test = pd.merge(test, fold4, on="id",how="left")

test["is_iceberg"] = test.mean(axis=1)

test[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg03.folds1-4.mean.csv", index=False)

fold1 = pd.read_csv(inDir + "/submissions/Prav.vgg04.fold1-test.csv")
fold2 = pd.read_csv(inDir + "/submissions/Prav.vgg04.fold2-test.csv")
fold3 = pd.read_csv(inDir + "/submissions/Prav.vgg04.fold3-test.csv")
fold4 = pd.read_csv(inDir + "/submissions/Prav.vgg04.fold4-test.csv")

test = pd.merge(fold1, fold2, on="id",how="left")
test = pd.merge(test, fold3, on="id",how="left")
test = pd.merge(test, fold4, on="id",how="left")

test["is_iceberg"] = test.mean(axis=1)

test[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg04.folds1-4.mean.csv", index=False)

vgg01_model = pd.read_csv(inDir + "/submissions/Prav.vgg03.folds1-4.mean.csv")
vgg02_model = pd.read_csv(inDir + "/submissions/Prav.vgg04.folds1-4.mean.csv")

models = pd.merge(vgg01_model, vgg02_model, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y"]].corr()

models["is_iceberg"] = models.mean(axis=1)

models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg03.vgg04.mean.csv", index=False)

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg03.vgg04.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/200_ens_densenet.csv")

models = pd.merge(model_01, model_02, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y"]].corr()

models["is_iceberg"] = models.mean(axis=1)

models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg0304_refDense.mean.csv", index=False)

################################################################################################################################

fold1 = pd.read_csv(inDir + "/submissions/Prav.vgg19_01.fold1-test.csv")
fold2 = pd.read_csv(inDir + "/submissions/Prav.vgg19_01.fold2-test.csv")
fold3 = pd.read_csv(inDir + "/submissions/Prav.vgg19_01.fold3-test.csv")
fold4 = pd.read_csv(inDir + "/submissions/Prav.vgg19_01.fold4-test.csv")

test = pd.merge(fold1, fold2, on="id",how="left")
test = pd.merge(test, fold3, on="id",how="left")
test = pd.merge(test, fold4, on="id",how="left")

test["is_iceberg"] = test.mean(axis=1)

test[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg19_01.folds1-4.mean.csv", index=False)

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg03.vgg04.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/200_ens_densenet.csv")
model_03 = pd.read_csv(inDir + "/submissions/Prav.vgg19_01.folds1-4.mean.csv")

models = pd.merge(model_01, model_02, on="id",how="left")
models = pd.merge(models, model_03, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y","is_iceberg"]].corr()

models.head()
models["is_iceberg"] = models.mean(axis=1)
models.head()
models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg0304_refDense_vgg19.mean.csv", index=False)

################################################################################################################################

################################################################################################################################

fold1 = pd.read_csv(inDir + "/submissions/Prav.vgg19_02.fold1-test.csv")
fold2 = pd.read_csv(inDir + "/submissions/Prav.vgg19_02.fold2-test.csv")
fold3 = pd.read_csv(inDir + "/submissions/Prav.vgg19_02.fold3-test.csv")
fold4 = pd.read_csv(inDir + "/submissions/Prav.vgg19_02.fold4-test.csv")

test = pd.merge(fold1, fold2, on="id",how="left")
test = pd.merge(test, fold3, on="id",how="left")
test = pd.merge(test, fold4, on="id",how="left")

test["is_iceberg"] = test.mean(axis=1)

test[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg19_02.folds1-4.mean.csv", index=False)

################################################################################################################################


################################################################################################################################

fold1 = pd.read_csv(inDir + "/submissions/Prav.vgg19_03.fold1-test.csv")
fold2 = pd.read_csv(inDir + "/submissions/Prav.vgg19_03.fold2-test.csv")
fold3 = pd.read_csv(inDir + "/submissions/Prav.vgg19_03.fold3-test.csv")
fold4 = pd.read_csv(inDir + "/submissions/Prav.vgg19_03.fold4-test.csv")

test = pd.merge(fold1, fold2, on="id",how="left")
test = pd.merge(test, fold3, on="id",how="left")
test = pd.merge(test, fold4, on="id",how="left")

test["is_iceberg"] = test.mean(axis=1)

test[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg19_03.folds1-4.mean.csv", index=False)

################################################################################################################################

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg19_02.folds1-4.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/Prav.vgg19_03.folds1-4.mean.csv")

models = pd.merge(model_01, model_02, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y"]].corr()

models["is_iceberg"] = models.mean(axis=1)

models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg19_0203.mean.csv", index=False)

################################################################################################################################

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg03.vgg04.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/Prav.vgg19_0203.mean.csv")

models = pd.merge(model_01, model_02, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y"]].corr()

models["is_iceberg"] = models.mean(axis=1)

models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg16_0304_vgg19_0203.mean.csv", index=False)

################################################################################################################################

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg0304_refDense.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/LB1541_final_ensemble.csv")

models = pd.merge(model_01, model_02, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y"]].corr()

models["is_iceberg"] = models.mean(axis=1)

models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg0304_refDense_ref2.mean.csv", index=False)

################################################################################################################################

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg03.folds1-4.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/Prav.vgg04.folds1-4.mean.csv")
model_03 = pd.read_csv(inDir + "/submissions/Prav.vgg19_02.folds1-4.mean.csv")
model_04 = pd.read_csv(inDir + "/submissions/Prav.vgg19_03.folds1-4.mean.csv")

models = pd.merge(model_01, model_02, on="id",how="left")
models = pd.merge(models, model_03, on="id",how="left")
models = pd.merge(models, model_04, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y","is_iceberg_x","is_iceberg_y"]].corr()

models.head()
models["is_iceberg"] = models.mean(axis=1)
models.head()
models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg16_0304_vgg19_0203.mean.csv", index=False)

################################################################################################################################

################################################################################################################################

model_01 = pd.read_csv(inDir + "/submissions/Prav.vgg16_0304_vgg19_0203.mean.csv")
model_02 = pd.read_csv(inDir + "/submissions/200_ens_densenet.csv")
model_03 = pd.read_csv(inDir + "/submissions/LB1541_final_ensemble.csv")
model_04 = pd.read_csv(inDir + "/submissions/vggbnw_fcn_en.csv")

models = pd.merge(model_01, model_02, on="id",how="left")
models = pd.merge(models, model_03, on="id",how="left")
models = pd.merge(models, model_04, on="id",how="left")

models[["is_iceberg_x","is_iceberg_y","is_iceberg_x","is_iceberg_y"]].corr()

models.head()
models["is_iceberg"] = models.mean(axis=1)
models.head()
models[["id","is_iceberg"]].to_csv(inDir+"/submissions/Prav.vgg16_0304_vgg19_0203.mean_3ref_mean.csv", index=False)

################################################################################################################################

File: Statoil/8000.Prav_ModelEnsembling.R

require.packages <- c(
  "data.table",
  "Matrix",
  "xgboost",
  "sqldf",
  "plyr",
  "dplyr",
  "ROCR",
  "Metrics",
  "pROC",
  "caret",
  "readr",
  "moments",
  "forecast",
  "ggplot2",
  "ggpmisc",
  "arules",
  "arulesViz",
  "extraTrees",
  "ranger",
  "randomForest",
  "knitr",
  "rmarkdown",
  "rJava",
  "tm"
)


##########################################################################################################################
# Function to install and load required packages
##########################################################################################################################
install.missing.packages <- function(x) {
  for (i in x) {
    #  require returns TRUE invisibly if it was able to load package
    if (!require(i , character.only = TRUE)) {
      #  If package was not able to be loaded then re-install
      install.packages(i , dependencies = TRUE)
      #  Load package after installing
      require(i , character.only = TRUE)
    }
  }
}

##########################################################################################################################
# Try function to install and load packages
##########################################################################################################################

install.missing.packages(require.packages)



fold1 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.nn01.fold1-test.csv")
fold2 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.nn01.fold2-test.csv") 
fold3 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.nn01.fold3-test.csv")
fold4 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.nn01.fold4-test.csv")

test <- left_join(fold1, fold2, by="id")
test <- left_join(test, fold3, by="id")
test <- left_join(test, fold4, by="id")

test$is_iceberg <- (test$is_iceberg.x + test$is_iceberg.y + test$is_iceberg.x.x + test$is_iceberg.y.y ) /4

cols <- c("id","is_iceberg")

write.csv(test[,cols],"C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.nn01.fold1-4-test.csv", row.names = FALSE, quote = FALSE)


model01 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.xgb001.fold1-4-test.csv") 
model02 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.nn01.fold1-4-test.csv") 

all_ensemble <- left_join(model01, model02, by = "id")
head(all_ensemble)
ensemble.features <- setdiff(names(all_ensemble),"id")
cor(all_ensemble[, ensemble.features])

all_ensemble$is_iceberg   <- 0.5 * all_ensemble$is_iceberg.x+ 0.5 * all_ensemble$is_iceberg.y

cols <- c("id","is_iceberg")

Ensemble <- all_ensemble[, cols]

head(all_ensemble)
head(Ensemble)


write.csv(Ensemble, 'C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav_xgb01_nn01_Mean.csv', row.names=FALSE, quote = FALSE)

x2 <- pi * 100^(-1:3)
round(x2, 3)
signif(x2, 3)



fold1 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.xgb001.fold1-test.csv")
fold2 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.xgb001.fold2-test.csv") 
fold3 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.xgb001.fold3-test.csv")
fold4 <- read_csv("C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.xgb001.fold4-test.csv")

test <- left_join(fold1, fold2, by="id")
test <- left_join(test, fold3, by="id")
test <- left_join(test, fold4, by="id")

test$is_iceberg <- (test$is_iceberg.x + test$is_iceberg.y + test$is_iceberg.x.x + test$is_iceberg.y.y ) /4

cols <- c("id","is_iceberg")

write.csv(test[,cols],"C:/Users/SriPrav/Documents/R/35Statoil/submissions/Prav.xgb001.fold1-4-test.csv", row.names = FALSE, quote = FALSE)

